src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1398,15 +1404,57 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
+            buildResult(displayClassInstancesInside, inScopeHoistedLocalSlots, parametersAreOutside ? ImmutableArray<string>.Empty : parameterNamesInOrder, displayClassVariablesBuilder, displayClassVariableNamesInsideInOrderBuilder);
+
+            // Let's add captured Primary Constructor parameters.
+            bool checkForPrimaryConstructor = true;
+
+            if (!currentFrame.IsStatic && isPrimaryConstructor)
+            {
+                checkForPrimaryConstructor = !tryAddCapturedPrimaryConstructorParameters(currentFrame, shadowingParameterNames: ImmutableArray<string>.Empty,
+                                                                                         possiblyCapturingType: currentFrame.ContainingType,
+                                                                                         possiblyCapturingTypeInstance: (Instance: null, Fields: ConsList<FieldSymbol>.Empty),
+                                                                                         displayClassVariablesBuilder, displayClassVariableNamesInsideInOrderBuilder);
+            }
+
+            buildResult(displayClassInstancesOutside, inScopeHoistedLocalSlots, parametersAreOutside ? parameterNamesInOrder : ImmutableArray<string>.Empty, displayClassVariablesBuilder, displayClassVariableNamesOutsideInOrderBuilder);
+
+            // ExtendBinderChain will place Primary Constructor parameters added below below InContainerBinder, rather than above it.
+            // However, since they are captured, they were not shadowed by any member at compile time.
+            // In theory, a shadowing member could be added into a base class after the build,
+            // but it is probably fine to shadow that member in EE. The member could still be accessed
+            // with qualification, but there wouldn't be a way to access captured parameter
+            // if we were to shadow it.
+            if (!isPrimaryConstructor && checkForPrimaryConstructor && currentFrame == currentSourceMethod && !currentFrame.IsStatic)
+            {
+                checkForPrimaryConstructor = !tryAddCapturedPrimaryConstructorParameters(currentFrame, shadowingParameterNames: parameterNamesInOrder,
+                                                                                         possiblyCapturingType: currentFrame.ContainingType,
+                                                                                         possiblyCapturingTypeInstance: (Instance: null, Fields: ConsList<FieldSymbol>.Empty),
+                                                                                         displayClassVariablesBuilder, displayClassVariableNamesOutsideInOrderBuilder);
+            }
+
+            if (checkForPrimaryConstructor && displayClassVariablesBuilder.Values.FirstOrDefault(v => v.Kind == DisplayClassVariableKind.This) is { } thisProxy)
```

Comment:
>&& [](http://example.com/codeflow?start=43&length=2)

Should the check include `&& !currentFrame.IsStatic`?

Reply:
> Should the check include ```&& !currentFrame.IsStatic```?

I don't think we care if the method itself is static as long as it has a "thisProxy" which can be supplied trough a parameter.

Reply:
All we care is whether we have access to `this` by whatever means